src/caret_analyze/common/type_check_decorator.py

```diff
@@ -246,14 +300,36 @@ def _custom_wrapper(*args, **kwargs):
 
                 if varargs_name is not None:
                     args = args[:arg_len] + _parse_collection_or_unpack(args[arg_len:])
+                sig = inspect.signature(func)
+                bound_args = sig.bind(*args, **kwargs)
+                args_dict = bound_args.arguments
                 return validate_arguments_wrapper(*args, **kwargs)
 
             except ValidationError as e:
+
                 loc_tuple = e.errors()[0]['loc']
                 annotations = get_annotations(func)
 
+                list_key = []
+                for key_value in args_dict.keys():
+                    list_key.append(key_value)
```

Comment:
nits:
The list of keys in the python dictionary can be obtained with the following code
```
list_key = list(args_dict.keys())
```

Reply:
Fixed